addons/mrp/models/mrp_bom.py

```diff
@@ -93,13 +93,25 @@ def onchange_routing_id(self):
     def name_get(self):
         return [(bom.id, '%s%s' % (bom.code and '%s: ' % bom.code or '', bom.product_tmpl_id.display_name)) for bom in self]
 
+    @api.model
+    def create(self, vals):
+        self._bom_find.clear_cache()
```

Comment:
You must add a docstring / comment saying why are you doing this. Something like: "Clear cache for getting proper `_bom_find` calls later`

Reply:
Done.